lib/krikri/entity_consumer.rb

```diff
@@ -1,15 +1,21 @@
 module Krikri
   ##
-  # A mixin for classes, like certain software agents, that generates entities.
-  # For example, a mapper usually generates RDF aggregations, such that
-  # Mapper::Agent includes EntityConsumer, and a mapper agent is instantiated
-  # with
-  #   a = Krikri::Mapper::Agent.new({
-  #     generator_uri: 'http://some.org/activity/1'
-  #   })
-  # such that
-  #   a.generator_activity
-  # returns a Krikri::Activity
+  # A mixin for `Krikri::SoftwareAgent`s that use entities. Provides a 
+  # mechanism for setting an `#entity_source` and consuming entities.
+  #
+  # For backwards compatability, this supports an older interface where entities
+  # are selected based on a `generator_activity`.
+  # 
+  # @example the deprecated interface
```

Comment:
I have an example of the new interface with the first alternative "entity source" in a later commit. Suggest bracketing this for the moment, since that PR is being fixed up now.